src/test/lib/operators/operator_performance_data_test.cpp

```diff
@@ -1,27 +1,307 @@
 #include <sstream>
+#include <vector>
+
+#include <magic_enum.hpp>
 
 #include "base_test.hpp"
-#include "operators/operator_performance_data.hpp"
+#include "expression/expression_functional.hpp"
+#include "hyrise.hpp"
+#include "operators/aggregate_hash.hpp"
+#include "operators/delete.hpp"
+#include "operators/get_table.hpp"
+#include "operators/join_hash.hpp"
+#include "operators/join_index.hpp"
+#include "operators/table_scan.hpp"
+#include "operators/table_wrapper.hpp"
+#include "storage/index/group_key/group_key_index.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
 
 namespace opossum {
 
-class OperatorsOperatorPerformanceDataTest : public BaseTest {
- public:
-  void SetUp() override {}
+class OperatorPerformanceDataTest : public BaseTest {
+ protected:
+  static void SetUpTestCase() {
+    _table = load_table("resources/test_data/tbl/int_int.tbl", 2);
+    _table_wrapper = std::make_shared<TableWrapper>(_table);
+    _table_wrapper->execute();
+  }
+
+  inline static std::shared_ptr<Table> _table;
+  inline static std::shared_ptr<TableWrapper> _table_wrapper;
 };
 
-TEST_F(OperatorsOperatorPerformanceDataTest, OutputToStream) {
-  auto performance_data = OperatorPerformanceData{};
+TEST_F(OperatorPerformanceDataTest, ElementsAreSet) {
+  const TableColumnDefinitions column_definitions{{"a", DataType::Int, false}};
+  auto table = std::make_shared<Table>(column_definitions, TableType::Data, 3);
+  table->append({1});
+  table->append({2});
+  table->append({3});
+
+  auto table_wrapper = std::make_shared<TableWrapper>(std::move(table));
+  table_wrapper->execute();
+  auto table_scan =
+      std::make_shared<TableScan>(table_wrapper, greater_than_(get_column_expression(table_wrapper, ColumnID{0}), 1));
+  table_scan->execute();
+
+  auto& performance_data = table_scan->performance_data;
+  EXPECT_TRUE(performance_data->executed);
+  EXPECT_TRUE(performance_data->has_output);
+  EXPECT_GT(performance_data->walltime.count(), 0ul);
+  EXPECT_EQ(performance_data->output_row_count, 2ul);
+}
+
+// Check for correct counting of skipped chunks/segment (this is different to chunk pruning, which happens within the
+// optimizer) due to early exists via the dictionary and counting of chunks/segments that are sorted and have thus been
+// scanned with a sorted search (i.e., binary search).
+TEST_F(OperatorPerformanceDataTest, TableScanPerformanceData) {
+  const TableColumnDefinitions column_definitions{{"a", DataType::Int, false}};
+  auto table = std::make_shared<Table>(column_definitions, TableType::Data, 2);
+  table->append({1});
+  table->append({2});
+  table->append({2});
+  table->append({3});
+
+  // Finalize last chunk
+  table->get_chunk(ChunkID{1})->finalize();
```

Comment:
Redundant comment. You could also use `get_last_chunk` to make your intentions clearer